exporter/datasetexporter/logs_exporter.go

```diff
@@ -65,6 +81,95 @@ func buildBody(attrs map[string]interface{}, value pcommon.Value) string {
 	return message
 }
 
+// Function maps OTel severity on the LogRecord to DataSet severity level (number)
+func mapOtelSeverityToDataSetSeverity(log plog.LogRecord) int {
+	// This function maps OTel severity level to DataSet severity levels
+	//
+	// Valid OTel levels - https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitynumber
+	// and valid DataSet ones - https://github.com/scalyr/logstash-output-scalyr/blob/master/lib/logstash/outputs/scalyr.rb#L70
+	sevNum := log.SeverityNumber()
+	sevText := log.SeverityText()
+
+	dataSetSeverity := defaultDataSetSeverityLevel
+
+	if sevNum > 0 {
+		dataSetSeverity = mapLogRecordSevNumToDataSetSeverity(sevNum)
+	} else if sevText != "" {
+		// Per docs, SeverityNumber is optional so if it's not present we fall back to SeverityText
+		// https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-severitytext
+		dataSetSeverity = mapLogRecordSeverityTextToDataSetSeverity(sevText)
+	}
+
+	// TODO: We should log in case we see invalid severity, but right now, afaik, we / OTEL
+	// don't have a concept of "rate limited" logging. We don't want to log every single
```

Comment:
Otel offers probabilistic head sampling for logs and traces, you can use that. There is no need for a TODO entry here as this is best captured with an issue on the github repository as a follow up instead.

Reply:
Per my reading and understanding of this sampling processor (https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/probabilisticsamplerprocessor), the processor is not flexible enough to only sample specific log messages.

So I could define a processor rule to sample all the log exporter logging messages, but not just this specific log message.

Other log messages are not that noisy / verbose so I would only want to sample this specific log message.

And I don't really want to use debug log level here either. I still want to log it, ideally, under warning log level, but only emit one message every X seconds in case there are lot of messages like that to prevent noise, overhead, etc.

Please correct me if I'm wrong / misunderstanding it.

Or is there some other existing code pattern / API in the code base I could use to do that?

Reply:
If not, to unblock the PR, I'm also fine with removing this TODO comment for now and open an issue instead.

Reply:
Please do.